include/swift/Frontend/FrontendOptions.h

```diff
@@ -310,6 +308,7 @@ class FrontendOptions {
   static bool canActionEmitInterface(ActionType);
 
 public:
+  static bool doesActionNeedSILOptimizer(ActionType);
```

Comment:
Bikeshedding: I don't like this name because it's not the optimizer itself that's interesting. `doesActionRunSILPasses`, maybe?